### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Slayer95**
- Regarding <code>nodemailer</code>: it still supports an API for transport options <i>alike</i> the old one, <strike>but I figure using <code>service</code> is a more useful default</strike>.

### CODE REVIEW

**Comment by Zarel** [comment]
How does the new nodemailer API work?

**Comment by Zarel** [comment]
PS Main server currently does use `nodemailer`, so we do need to know how to do it before merging

**Comment by Slayer95** [comment]
If we are using one of the services listed here https://github.com/andris9/nodemailer-wellknown, just set the <code>service</code> value to the appropriate one
Otherwise:
- The transport method defaults to SMTP (in that case, options should be like old <code>config.crashguardemail.options</code>, but
* <code>secureConnection</code> -> <code>secure</code>
* <code>maxConnections</code> always 1 (remove it)

**Comment by Zarel** [comment]
Well, fill it out with the "Otherwise" stuff. :|

**Comment by Slayer95** [comment]
Ok, done

File: app.js [hunk 1]
```diff
@@ -58,10 +58,11 @@ function runNpm(command) {
 
 try {
 	require('sugar');
+	require('es6-shim');
```
Comment (Zarel) [inline app.js]: We only use `Map` and `Set`, right? Let's just test for the existence of `global.Set` before deciding whether or not to shim it?

Comment (Slayer95) [inline app.js]: We also have some instances of new native prototype methods, like <code>String.prototype.startsWith</code>. I wouldn't really put my hands on fire for any Node version with <code>global.Set</code> also having every other ES6 feature we are using, but it should be fine.

Also, we aren't using <code>Set</code>, so in that case we would check for <code>global.Map</code>.

**Comment by Slayer95** [comment]
Ok, updated... Also removed the check for <code>Reflect</code> to be more aligned to the detection of needed features.

File: app.js [hunk 2]
```diff
@@ -56,12 +56,15 @@ function runNpm(command) {
...
+var isLegacyEngine = !global.Map;
+
 try {
 	require('sugar');
+	if (isLegacyEngine) require('es6-shim');
 } catch (e) {
 	runNpm('install --production');
 }
-if (!Object.select) {
+if (isLegacyEngine && !(new Map().set(0, 0) instanceof Map)) {
```
Comment (Zarel) [inline app.js]: Do we need this much? o_O

Comment (Slayer95) [inline app.js]: Not currently as the old <code>es6-shim</code> version fails this test, but the tournaments code would benefit in several places from chaining <code>Map#set</code<

**Comment by Slayer95** [comment]
What about this?

**Comment by Zarel** [comment]
Okay, I think I understand.

**Pull request merged**